services/crawling.js

```diff
@@ -35,14 +36,16 @@ const getPostCrawlingData = async (post) => {
   const likeCount = await page.evaluate(
     () => parseInt(document.querySelector(".u_cnt._count").innerText.trim()) || 0
   );
+  const isAd = await Promise.resolve(content.includes("소정의 원고료"));
```

Comment:
content에 게시물 본문이 담긴 후 광고 여부를 확인하기 위해 `Promise.resolve` 사용한 것 👍

Reply:
맞아요 ㅎ 비동기의 동기적(순차적) 처리를 위해서 `Promise.resolve()`를 활용했어요 👍

Reply:
```suggestion
  const validateAdKeyword = ["소정의 원고료", "지원 받아 작성한"];
  const isAd = await Promise.resolve(validateAdKeyword.some(keyword => content.includes(keyword)));
```
오 좋네요! 추가로 광고 검사 키워드를 여러개 등록하게 확장할 것을 고려해서, 위와 같이 배열로 관리해보면 어떨까요?
현재는 "소정의 원고료" 1개만 두었지만 게시물 마다 광고문구 워딩이 달라서 관련 키워드가 추가될 가능성이 높다고 생각해요.

해당 키워드를 하나라도 포함하면 true 반환하도록 `some` 메소드를 적용했는데, 테스트 통과해서 코드도 첨부 드려요!
그리고 `validateAdKeyword`는 `getKeywordPostList` 함수 밖에 선언되거나 상수로 관리되는게 매번 선언되지 않아서 효율적이겠습니다!

> The some() method of [Array](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array) instances tests whether at least one element in the array passes the test implemented by the provided function. It returns true if, in the array, it finds an element for which the provided function returns true; otherwise it returns false. It doesn't modify the array. [MDN - Array.prototype.some()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some)

아래 테스트는 제가 임의로 "단풍"을 배열 안 요소로 두었을 때, `some()` 메소드로 잘 잡는지 점검한 내역이에요.
`isAd`를 보면 true로 잘 선별됩니다.
<img width="356" alt="Screenshot 2024-11-13 at 10 51 59 AM" src="https://github.com/user-attachments/assets/fcacdadd-d031-4137-8409-f4177b98bcc8">

Reply:
네 앞으로 더 추가될 것을 고려하여 아래와 같이 수정하도록 하겠습니다.

```js
// validation.js

const validateAdKeyword = ["소정의 원고료", "소정의 수수료", "수수료를 지급받아", "원고료를 지급받아"];
```
```js
// crawling.js

const isAd = await Promise.resolve(validateAdKeyword.some((adKeyword) => content.includes(adKeyword)));
```